content/en/ebpf.md

```diff
@@ -0,0 +1,40 @@
+---
+title: eBPF
+status: Completed
+category: architecture
+---
+
+## What it is
+
+eBPF, or extended Berkeley Packet Filter, is a technology that allows small, sandboxed programs or scripts to run in the kernel space of a Linux system without having to change the kernel's source code or load Linux kernel modules.
+
+A Linux system has two spaces: the kernel and the user space. 
+The kernel represents the operating system's core and is the only part 
+with unlimited access to the hardware. 
+
+Applications reside in the user space, and when they need higher permissions, 
+they send a request to the kernel.
+For applications that require more flexibility, such as direct hardware 
+access, the kernel can be extended via what is known as the "Linux 
+kernel modules" approach. This approach extends the kernel's default functionality
+ allowing applications deeper access to the underlying components. 
+ Unfortunately, this could introduce security risks as well.
+
+## Problem it addresses
+Typically, applications run in the user space, and if the application requires some privileges from the kernel (e.g., to access some hardware), 
+it requests it from the kernel via a so-called "system call."  
+In most cases, this approach works just fine. However, there are instances where developers require more flexibility for low-level system access.
+Observability, security, and networking features are some of the ideal examples to be implemented within the kernel space.
+One of the possibilities to reach such a goal is to use Linux kernel modules, which allow extending the kernel base without adding directly to the kernel source code. 
```

Comment:
Making that sentence a little shorter but 100% sure what it means. Does it mean without modifying the kernel source code?
```suggestion
To achieve that, we can use Linux kernel modules, extending the kernel base without adding directly to the kernel source code. 
```

Reply:
As I understand it, there is specific code for the kernel module but it is not part of the main kernel source code. The kernel module could be loaded at runtime if needed and extend the functionality of the kernel.

Reply:
yes, this is correct. The suggestion does not change the statement, so should be fine.